### ISSUE TITLE
Empty payment amount `!=` payment status success

### ISSUE THREAD

**Issue by remcotolsma** [pronamic/wp-pronamic-pay#188]
Currently we give all payments with an empty (`€ 0`) amount the status success and return early so the payment is not started at the gateway: https://github.com/pronamic/wp-pay-core/blob/develop/src/Plugin.php#L1025-L1045.

This seems less convenient because subscriptions may include a free first period (trial), but with some payment methods / gateways a minimum payment is required to obtain a mandate.

This is the case for a recurring direct debit payment where a mandate is obtained via a first iDEAL payment. Certain gateways must have the chance to throw an exception in that case.

<img width="1893" alt="Schermafbeelding 2021-08-20 om 09 46 47" src="https://user-images.githubusercontent.com/869674/130199065-fff89ca3-882e-47eb-b003-325210c20f98.png">

<img width="1893" alt="Schermafbeelding 2021-08-20 om 09 47 03" src="https://user-images.githubusercontent.com/869674/130199076-f587fc9e-bc09-4dc9-b4c7-0d4b08a80f36.png">

It would be great if the Mollie gateway could throw an exception in case of an `€ 0` payment for the 'Direct Debit mandate via iDEAL payment method' (`DIRECT_DEBIT_IDEAL`).

https://help.mollie.com/hc/nl/articles/214072489-Hoe-verkrijg-ik-een-mandaat-van-mijn-klant-customer-voor-Recurring-

I also noted a `$maybe_tokenize` variable introduced in https://github.com/pronamic/wp-pay-core/commit/8d90f0dba9f3c6ac0d95bf28ffe6598168b54dca that may also be related to this.

_Internal HelpScout ticket:_ https://secure.helpscout.net/conversation/1604745461/22551?folderId=1425710

**Comment by rvdsteege** [comment]
I agree we need to improve on this. We'll be able to set payment method constraints (like minimum amount) with the updated Payment Methods API, which will resolve this issue.

- https://github.com/pronamic/wp-pronamic-pay/issues/154

**Comment by rvdsteege** [comment]
Further investigation shows that subscriptions for MemberPress memberships with an invalid initial amount for `DIRECT_DEBIT_IDEAL` are already impossible to start and result in a gateway error "The amount is lower than the minimum". The status of the first payment gets updated to **Failed**.

<img width="646" alt="Schermafbeelding 2021-08-23 om 11 18 10" src="https://user-images.githubusercontent.com/10371164/130423172-04a15d12-1f28-4677-ac0e-5a81036a2c63.png">

In the past, if the first payment failed, the subscription status was updated to **Cancelled** to prevent unintended reactivation of subscription (if a mandate would become available at a later time, for example if a new subscription had been started successfully by the same user). It seems this has been removed some time ago:

- https://github.com/pronamic/wp-pay-core/commit/b26ec8237bb0df23c5ea3153c0e1ca382d2667d4

Therefore, recurring payments are still created for these failed subscriptions and the recurring payments result in the "No suitable mandates found for customer" error. For a failed first payment, the subscription should be canceled or put on hold to prevent further payments for the invalid subscription. The error about the required minimum amount should already be sufficient to prevent creating invalid subscriptions, as we can assume this error will be noticed by users during setup testing.

Thoughts, @remcotolsma?
